akka-javasdk/src/main/scala/akka/javasdk/impl/consumer/ConsumerImpl.scala

```diff
@@ -153,7 +164,9 @@ private[impl] final class MessageContextImpl(
     override val metadata: Metadata,
     timerClient: TimerClient,
     tracerFactory: () => Tracer,
-    val span: Option[Span])
+    val span: Option[Span],
+    selfRegion: String,
```

Comment:
In the issue there is another example use case

> call an external service, internal component, or publish to message broker from one region, ignore in other regions

For that we would have to expose the selfRegion.

Reply:
I thought about adding it to the creation context instead of on the message.